packages/ui/src/views/repo/repo-summary/components/summary.tsx

```diff
@@ -0,0 +1,135 @@
+import { useNavigate } from 'react-router-dom'
+
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+  ButtonGroup,
+  Icon,
+  Spacer,
+  StackedList,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Text
+} from '@/components'
+import { CommitCopyActions } from '@/components/commit-copy-actions'
+import { getInitials } from '@/utils/utils'
+
+import { FileProps, SummaryItemType } from '../repo-summary.types'
+
+interface SummaryProps {
+  latestFile: Pick<FileProps, 'user' | 'lastCommitMessage' | 'timestamp' | 'sha'>
+  files?: FileProps[]
+}
+
+const TopTitle = ({ file }: { file: Pick<FileProps, 'user' | 'lastCommitMessage' | 'timestamp' | 'sha'> }) => {
+  const { user, lastCommitMessage } = file
+
+  return (
+    <ButtonGroup.Root verticalAlign="center" spacing="2">
+      <Avatar size="6">
+        <AvatarImage src={user?.avatarUrl} />
+        <AvatarFallback>
+          <Text size={0} color="tertiaryBackground">
+            {getInitials(user?.name || '')}
+          </Text>
+        </AvatarFallback>
+      </Avatar>
+      <Text size={2} weight="normal" color="tertiaryBackground" wrap="nowrap">
+        {user?.name}
+      </Text>
+      <Text size={2} weight="normal" color="primary" className="line-clamp-1">
+        {lastCommitMessage}
+      </Text>
+    </ButtonGroup.Root>
+  )
+}
+
+const TopDetails = ({ file }: { file: Pick<FileProps, 'user' | 'lastCommitMessage' | 'timestamp' | 'sha'> }) => {
+  const { sha, timestamp } = file
+  return (
+    <ButtonGroup.Root verticalAlign="center" spacing="2">
+      <CommitCopyActions sha={sha || ''} />
+      <Text size={2} weight="normal" color="tertiaryBackground">
+        |
+      </Text>
+      <Text size={2} weight="normal" color="tertiaryBackground">
+        {timestamp}
+      </Text>
+    </ButtonGroup.Root>
+  )
+}
+
+const Summary: React.FC<SummaryProps> = ({ latestFile, files }) => {
+  const navigate = useNavigate()
+
+  return (
+    <>
+      <StackedList.Root>
+        <StackedList.Item disableHover isHeader className="px-3 py-2.5">
+          {latestFile ? (
+            <>
+              <StackedList.Field title={<TopTitle file={latestFile} />} />
+              <StackedList.Field right title={<TopDetails file={latestFile} />} />
+            </>
+          ) : (
+            <Text>No files available</Text>
+          )}
+        </StackedList.Item>
+      </StackedList.Root>
+      <Spacer size={5} />
+      <Table variant="asStackedList">
+        <TableHeader>
+          <TableRow>
+            <TableHead>Name</TableHead>
+            <TableHead>Last commit message</TableHead>
+            <TableHead className="text-right">Date</TableHead>
+          </TableRow>
+        </TableHeader>
+        {files && files.length > 0 ? (
```

Comment:
Not an issue, but you can simplify a wee bit

```suggestion
        {!!files?.length ? (
```

Reply:
Unfortunately, in this case, it will throw an error due to the double negation.
<img width="493" alt="Cursor 2024-11-28 23 17 06" src="https://github.com/user-attachments/assets/34f0dca9-492d-4241-95f7-3177581e3ed9">

Therefore, I think we can simplify it even further to just `files?.length`

Reply:
Absolutely 👍🏻